extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js

```diff
@@ -14,6 +26,63 @@ export class AmpStoryShoppingAttachment extends AmpStoryPageAttachment {
 
     /** @private {?Element} */
     this.container_ = null;
+
+    /** @private @const {?../../amp-story/1.0/amp-story-store-service.AmpStoryStoreService} */
+    this.storeService_ = null;
+  }
+
+  /**
+   * @private
+   * @return {?Element} the parent amp-story-page
+   */
+  getPage_() {
+    return closest(
+      dev().assertElement(this.element),
+      (el) => el.tagName.toLowerCase() === 'amp-story-page'
+    );
+  }
+
+  /**
+   * Updates the CTA based on the shopping data.
+   * @param {!ShoppingDataDef} shoppingData
+   * @private
+   */
+  updateCtaText_(shoppingData) {
+    const pageIdElem = this.getPage_();
+
+    const shoppingTags = Array.from(
+      pageIdElem.getElementsByTagName('amp-story-shopping-tag')
+    ).filter((tag) => shoppingData[tag.getAttribute('data-tag-id')]);
+
+    const ctaButton = pageIdElem
+      .querySelector('.i-amphtml-story-page-open-attachment-host')
+      ?.shadowRoot.querySelector('.i-amphtml-story-page-open-attachment');
+
+    const i18nString =
+      shoppingTags.length === 1
+        ? localize(
+            this.element,
+            LocalizedStringId_Enum.AMP_STORY_SHOPPING_SHOP_LABEL
+          ) +
+          ' ' +
+          this.storeService_.get(StateProperty.SHOPPING_DATA)[
+            shoppingTags[0].getAttribute('data-tag-id')
+          ]['product-title']
+        : localize(
+            this.element,
+            LocalizedStringId_Enum.AMP_STORY_SHOPPING_VIEW_ALL_PRODUCTS
+          );
+
+    if (!ctaButton) {
```

Comment:
We should add a comment here like:
"If CTA button isn't built, set the data-cta-text so text is added when it's built."